### DISCUSSION TITLE
Download attachment part ID

### DISCUSSION THREAD

**Discussion by dobainier** [umanamente/n8n-nodes-imap#13] [Q&A]
Hello, thank you very much for your node, it's really great.
I have a problem when i try to download an attachment, I need to specify the Attachment Part ID :
- There always the ID 2, because it's the first attachment, but sometimes there's more attachment with emails. So sometimes I need also the ID's 3, 4, 5....
- But if a mail has only one attachment it doesn't work if I put multiple Attachment Part ID
- The ugly workaround I found is a Switch node before the download IMAP node to check how many attachments there is, and it routes to multiple Download IMAP with different ID's...

I'm sure there is a better way with an expression in Attachment Part ID, but I'm not a coder so I don't get it.

Somebody can help me ?

Thank you very much.

**Comment by umanamente** [comment]
In "Get Many Emails" node there is a "`Include Message Parts`" parameter with "`Attachments info`" option. If you enable it, it will add a list of attachments to every email info.

**Reply by umanamente** [reply]
![2024-04-18 16_04_34](https://github.com/umanamente/n8n-nodes-imap/assets/10637425/ca66d29f-e31d-42b0-8d6c-044cf236da1d)
